cmake/org.eclipse.cdt.cmake.ui/src/org/eclipse/cdt/cmake/ui/internal/CMakePreferencePage.java

```diff
@@ -205,7 +373,121 @@ public boolean performOk() {
 		filesToAdd.clear();
 		filesToRemove.clear();
 
+		// Update Preferences for cmakeSupplier
+		try {
+			getPreferences().clear();
+			getPreferences().node(CMakeBuildEnvironmentSupplier.CMAKE_GENERATOR_LOCATION).clear();
+			getPreferences().putBoolean(CMakeBuildEnvironmentSupplier.ENABLE_USE_CMAKE_LOCATION, useCmakeToolLocation);
+			if (!cmakeLocation.isEmpty()) {
+				getPreferences().put(CMakeBuildEnvironmentSupplier.CMAKE_LOCATION, cmakeLocation);
+			}
+			int index;
+			for (index = 0; index < generatorLocations.length; index++) {
+				getPreferences().node(CMakeBuildEnvironmentSupplier.CMAKE_GENERATOR_LOCATION).put(
+						String.format(CMakeBuildEnvironmentSupplier.LOCATION_NODE, index), generatorLocations[index]);
+
+			}
+			getPreferences().flush();
+		} catch (BackingStoreException e) {
+			Activator.log(e);
+		}
 		return true;
 	}
 
+	private void updateCmakeToolGroupData() {
+		try {
+			useCmakeToolLocation = getPreferences().getBoolean(CMakeBuildEnvironmentSupplier.ENABLE_USE_CMAKE_LOCATION,
+					false);
+			cmakeLocation = getPreferences().get(CMakeBuildEnvironmentSupplier.CMAKE_LOCATION,
+					CMakeBuildEnvironmentSupplier.EMPTY_STRING);
+			List<String> genlocs = new ArrayList<>();
+			String[] keys = getPreferences().node(CMakeBuildEnvironmentSupplier.CMAKE_GENERATOR_LOCATION).keys();
+			int index;
+			for (index = 0; index < keys.length; index++) {
+				genlocs.add(getPreferences().node(CMakeBuildEnvironmentSupplier.CMAKE_GENERATOR_LOCATION).get(
+						String.format(CMakeBuildEnvironmentSupplier.LOCATION_NODE, index),
+						CMakeBuildEnvironmentSupplier.EMPTY_STRING));
+			}
+			generatorLocations = genlocs.toArray(new String[0]);
```

Comment:
You can use more recent Java way to avoid having to resize the array:

```suggestion
			generatorLocations = genlocs.toArray(String[]::new);
```